server/controllers/contest.js

```diff
@@ -60,3 +62,45 @@ exports.getContests = asyncHandler(async (req, res, next) => {
     throw new Error('failed to get contests');
   }
 });
+
+// handler for create submission by contest id
+exports.createSubmissionByContestId = asyncHandler(async (req, res, next) => {
+  const contestId = req.params.id;
+  const s3UrlArray = req.body.data;
+  const userId = req.user.id;
+  if (!ObjectId.isValid(contestId)) {
+    return res.status(400).json({
+      error: 'Contest ID is invalid.',
+    });
+  }
+  try {
+    const userExist = await Submission.findOne({ userId: userId, contestId: contestId });
+    if (userExist) {
+      const submissionData = await Submission.findByIdAndUpdate(
+        userExist._id,
+        { files: s3UrlArray },
+        {
+          new: true,
+        },
+      );
+      res.status(200).json({
+        submission: submissionData,
+      });
+    } else {
+      const submissionData = await Submission.create({
+        contestId: contestId,
+        userId: userId,
+        files: s3UrlArray,
+        isActive: true,
+      });
+      if (submissionData) {
+        res.status(200).json({
```

Comment:
Please use appropriate status codes for resource creation and server error.

Ref: https://www.restapitutorial.com/httpstatuscodes.html

Reply:
thank you for the ref website, and I changed to code 201 for create success and code 500 for server error.